modules/oe_webtools_analytics/tests/src/Kernel/ConfigurationTest.php

```diff
@@ -36,23 +35,29 @@ public function testLibraryLoading(): void {
       ->set("instance", "testing");
     $config->save();
 
-    foreach (Cache::getBins() as $service_id => $cache_backend) {
-      if ('dynamic_page_cache' === $service_id || 'page' === $service_id) {
-        $cache_backend->deleteAll();
-      }
-    }
-
     $this->drupalGet('<front>');
     $this->assertSession()
       ->responseContains('<script type="application/json">{"utility":"piwik","siteID":"123","sitePath":["ec.europa.eu"],"instance":"testing"}</script>');
 
+    // Test the cache invalidation.
+    $config = \Drupal::configFactory()
+      ->getEditable(AnalyticsEventInterface::CONFIG_NAME)
+      ->set("siteID", "1234")
+      ->set("sitePath", "ec.europa.eu")
+      ->set("instance", "testing");
+    $config->save();
+
+    $this->drupalGet('<front>');
+    $this->assertSession()
+      ->responseContains('<script type="application/json">{"utility":"piwik","siteID":"1234","sitePath":["ec.europa.eu"],"instance":"testing"}</script>');
+
     $this->drupalGet('not-existing-page');
     $this->assertSession()
-      ->responseContains('<script type="application/json">{"utility":"piwik","siteID":"123","sitePath":["ec.europa.eu"],"is404":true,"instance":"testing"}</script>');
+      ->responseContains('<script type="application/json">{"utility":"piwik","siteID":"1234","sitePath":["ec.europa.eu"],"is404":true,"instance":"testing"}</script>');
 
     $this->drupalGet('admin');
     $this->assertSession()
-      ->responseContains('<script type="application/json">{"utility":"piwik","siteID":"123","sitePath":["ec.europa.eu"],"is403":true,"instance":"testing"}</script>');
+      ->responseContains('<script type="application/json">{"utility":"piwik","siteID":"1234","sitePath":["ec.europa.eu"],"is403":true,"instance":"testing"}</script>');
```

Comment:
To understand better the test, we could put this test before to change the configuration.